acceptance-tests/bosh_helpers.go

```diff
@@ -0,0 +1,250 @@
+package acceptance_tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	"gopkg.in/yaml.v2"
+)
+
+type haproxyInfo struct {
+	SSHPrivateKey string
+	SSHUser       string
+	PublicIP      string
+}
+
+type varsStoreReader func(interface{}) error
+
+// Helper method for deploying HAProxy
+// Takes the HAProxy backend port, an array of custom opsfiles, a map of custom vars
+// Returns 'info' struct containing public IP and ssh creds, and a callback to deserialise properties from the vars store
+func deployHAProxy(haproxyBackendPort int, customOpsfiles []string, customVars map[string]interface{}) (haproxyInfo, varsStoreReader) {
+	sshUser := "ginkgo"
+
+	opsfileChangeVersion := `---
```

Comment:
They could be but the structure of the creds struct that comes after that depends on the contents of the ops file, and these are different for each test. As the `var creds struct` is tightly-coupled to the contents of the opsfile I thought it better to have them in the same place (otherwise we risk changes to one not being applied in the other and vice-versa)